### PR TITLE
set rpath for external libsemigroups

### PR DESCRIPTION
**Pull request by dimpase**
set rpath for external `libsemigroups`, so that `LD_LIBRARY_PATH` is not needed if `libsemigroups` is installed to a non-standard location

### CODE REVIEW

**Comment by dimpase** [comment]
I've left `m4/ax_check_libsemigroups.m4` largely as it was, no refactoring. I only split to very long strings into two, and
added the necessary stuff.

**Comment by james-d-mitchell** [comment]
Thanks @dimpase

**Review by james-d-mitchell — APPROVED** [review]
This seems to do exactly what is desired, happy to approve. @fingolfin could you please also have a quick look at this if you have the time?

File: m4/ax_check_libsemigroup.m4 [hunk 1]
```diff
@@ -15,7 +15,8 @@ AC_DEFUN([AX_CHECK_LIBSEMIGROUPS], [
                           [libsemigroups >= $REQUI_LIBSEMIGROUPS_VERSION],
 			  [need_included_libsemigroups=no],
 			  [need_included_libsemigroups=yes])],
-	[AC_MSG_NOTICE([ignoring flag --with-external-libsemigroups, the Semigroups configure file was created on a system without m4 macros for pkg-config available...])])
+	[AC_MSG_NOTICE([ignoring flag --with-external-libsemigroups, the Semigroups configure file
+                        was created on a system without m4 macros for pkg-config available...])])
```
Comment (fingolfin) [inline m4/ax_check_libsemigroup.m4:19]: I think this should be undone, it causes the error to change from
```
...
checking for awk... awk
configure: using included libsemigroups...
configure: error: libsemigroups is required, clone or download the repo from https://github.com/libsemigroups/libsemigroups into this directory
```
which is sensibly wrapped in my terminal, to
```
checking for awk... awk
configure: using included libsemigroups...
configure: error: libsemigroups is required, clone or download the repo from
                                  https://github.com/libsemigroups/libsemigroups into this directory
```
which is rather weird.

File: m4/ax_check_libsemigroup.m4 [hunk 2]
```diff
@@ -27,7 +28,8 @@ AC_DEFUN([AX_CHECK_LIBSEMIGROUPS], [
    		[AS_IF(
    		   [test -f libsemigroups/include/libsemigroups.hpp],
                    [],
-                   [AC_MSG_ERROR([libsemigroups is required, clone or download the repo from https://github.com/libsemigroups/libsemigroups into this directory])])])
+                   [AC_MSG_ERROR([libsemigroups is required, clone or download the repo from
+                                  https://github.com/libsemigroups/libsemigroups into this directory])])])
```
Comment (fingolfin) [inline m4/ax_check_libsemigroup.m4:32]: Should also be undone, IMHO

File: m4/ax_check_libsemigroup.m4 [hunk 3]
```diff
@@ -53,6 +55,8 @@ AC_DEFUN([AX_CHECK_LIBSEMIGROUPS], [
   else 
         LIBSEMIGROUPS_VERSION="$(pkg-config --modversion libsemigroups)"
 	AC_MSG_NOTICE([using external libsemigroups $LIBSEMIGROUPS_VERSION])
```
[the remark this answers is not in the document]
Comment (fingolfin) [reply m4/ax_check_libsemigroup.m4:57]: (Indeed, this whole file mixes tabs and spaces liberally. Probably best to ignore this for this PR)

File: m4/ax_check_libsemigroup.m4 [hunk 4]
```diff
@@ -53,6 +55,8 @@ AC_DEFUN([AX_CHECK_LIBSEMIGROUPS], [
...
+        PKG_CHECK_VAR([LIBSEMIGROUPS_RPATH], [libsemigroups], [libdir],
+              [AC_SUBST([LIBSEMIGROUPS_RPATH],[-Wl,-rpath,${LIBSEMIGROUPS_RPATH}])])
```
Comment (fingolfin) [inline m4/ax_check_libsemigroup.m4:59]: That's the core of the change. Looks OK to me.

File: Makefile.gappkg [hunk 5]
```diff
@@ -138,7 +138,7 @@ gen/%.$(GAP_OBJEXT): %.s Makefile
 # build rule for linking all object files together into a kernel extension
 $(KEXT_SO): $(KEXT_OBJS)
 	@mkdir -p $(@D)
-	$(QUIET_GAC)$(GAP_CXX) -o $@ $(GAP_LDFLAGS) $(GAC_LDFLAGS) $(KEXT_OBJS) $(KEXT_LDFLAGS)
+	$(QUIET_GAC)$(GAP_CXX) -o $@ $(GAP_LDFLAGS) $(GAC_LDFLAGS) $(KEXT_OBJS) $(KEXT_LDFLAGS) $(LIBSEMIGROUPS_RPATH)
```
Comment (fingolfin) [inline Makefile.gappkg:141]: I'd say this should rather be added to `KEXT_LDFLAGS`, that'll be better in the long run. But for now, either works *shrug*

**Review by fingolfin — COMMENTED** [review]
Thanks! The principle seems fine to me, but I have some minor remarks on details.

**Comment by james-d-mitchell** [comment]
Thanks @fingolfin, @dimpase I just rebased onto `main` I can make Max's suggested changes if you want?

File: m4/ax_check_libsemigroup.m4 (hunk 1 above)
Comment (james-d-mitchell) [reply m4/ax_check_libsemigroup.m4:19]: Agreed, I think this change should be undone.

Comment (dimpase) [reply m4/ax_check_libsemigroup.m4:19]: There are no two equal terminals, besides, such long lines of code aren't helping.
I can try to make sure the output is formatted better.

**Comment by dimpase** [comment]
> Thanks @fingolfin, @dimpase I just rebased onto `main` I can make Max's suggested changes if you want?

It's up to you. I'd rather move the 2nd lines of these long error messages a bit to the left,
to reduce the unfortunate indentation.

**Comment by dimpase** [comment]
You can also wrap the messages in `m4_normalize()`, this will remove any new lines and extra spaces/tabs, I think - but keep short lines in the source.

**Comment by fingolfin** [comment]
which terminals do you know that do not wrap long too lines?

In any case, I am sure the message can be improved, but the current form in this PR isn't an improvement anywhere: on terminals that do wrap around it is worse, on terminals that don't it would also be worse, as they still couldn't see most of the second line.

File: m4/ax_check_libsemigroup.m4 (hunk 1 above)
Comment (fingolfin) [reply m4/ax_check_libsemigroup.m4:19]: Have you tested the message? It was perfectly legible for me before your changes but not anymore afterwards. Was the experience different for you?

**Comment by dimpase** [comment]
I think with the change I just pushed they will be again printed as one line - but split in the source.

File: m4/ax_check_libsemigroup.m4 (hunk 1 above)
Comment (dimpase) [reply m4/ax_check_libsemigroup.m4]: I changed it to avoid extra-long lines in the source, most of all.

**Review by fingolfin — APPROVED** [review]
Looks good to me now, thanks!

**Pull request merged**